cmd/dep/godep_importer.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/dep"
+	fb "github.com/golang/dep/internal/feedback"
+	"github.com/golang/dep/internal/gps"
+	"github.com/pkg/errors"
+)
+
+const godepJSONName = "Godeps.json"
+
+type godepImporter struct {
+	json godepJSON
+
+	logger  *log.Logger
+	verbose bool
+	sm      gps.SourceManager
+}
+
+func newGodepImporter(logger *log.Logger, verbose bool, sm gps.SourceManager) *godepImporter {
+	return &godepImporter{
+		logger:  logger,
+		verbose: verbose,
+		sm:      sm,
+	}
+}
+
+type godepJSON struct {
+	Imports []godepPackage `json:"Deps"`
+}
+
+type godepPackage struct {
+	ImportPath string `json:"ImportPath"`
+	Rev        string `json:"Rev"`
+	Comment    string `json:"Comment"`
+}
+
+func (g *godepImporter) Name() string {
+	return "godep"
+}
+
+func (g *godepImporter) HasDepMetadata(dir string) bool {
+	y := filepath.Join(dir, "Godeps", godepJSONName)
+	if _, err := os.Stat(y); err != nil {
+		return false
+	}
+
+	return true
+}
+
+func (g *godepImporter) Import(dir string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	err := g.load(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return g.convert(pr)
+}
+
+func (g *godepImporter) load(projectDir string) error {
+	g.logger.Println("Detected godep configuration files...")
+	j := filepath.Join(projectDir, "Godeps", godepJSONName)
+	if g.verbose {
+		g.logger.Printf("  Loading %s", j)
+	}
+	jb, err := ioutil.ReadFile(j)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to read %s", j)
+	}
+	err = json.Unmarshal(jb, &g.json)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to parse %s", j)
+	}
+
+	return nil
+}
+
+func (g *godepImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from Godeps.json ...")
+
+	manifest := &dep.Manifest{
+		Constraints: make(gps.ProjectConstraints),
+	}
+	lock := &dep.Lock{}
+
+	for _, pkg := range g.json.Imports {
+		// ImportPath must not be empty
+		if pkg.ImportPath == "" {
+			err := errors.New("Invalid godep configuration, ImportPath is required")
+			return nil, nil, err
+		}
+
+		// Obtain ProjectRoot. Required for avoiding sub-package imports.
+		ip, err := g.sm.DeduceProjectRoot(pkg.ImportPath)
```

Comment:
Ah okay after scrolling down to the test, it makes sense. Glide lists unique packages and then lists the subpackages used separately which explains why I didn't have to check for this. 😁 Carry on!